geoapps/edge_detection/application.py

```diff
@@ -240,21 +260,32 @@ def update_name(self, _):
 
     def compute_trigger(self, _):
         param_dict = self.get_param_dict()
-        param_dict["geoh5"] = self.workspace
-
         if param_dict.get("objects", None) is None:
             return
 
-        with fetch_active_workspace(self.workspace):
-            new_params = EdgeDetectionParams(**param_dict)
+        with fetch_active_workspace(self.workspace, mode="r+"):
```

Comment:
```suggestion
        with fetch_active_workspace(self.workspace):
```
Shouldn't have to be in r+ since only reading data.

Reply:
`get_edges` has an `add_data` that fails if in `mode='r'`